src/api/apiProducts.ts

```diff
@@ -1,19 +1,49 @@
 import apiEndpoints from "@/api.endpoints";
 
 import handleErrors from "@/utils/handleErrors";
-import { Game } from "@/types";
-import { get } from "./requests";
+import { Game, Product } from "@/types";
+import { del, get, post, update } from "./requests";
 
 interface GetProductsParams {
   urlParams: string;
 }
 
-export default async function getProducts({ urlParams }: GetProductsParams): Promise<Game[]> {
+export async function getProducts({ urlParams }: GetProductsParams): Promise<Game[]> {
   try {
     const data = await get(`${apiEndpoints.getProducts}?${urlParams}`);
     return data as Game[];
   } catch (error) {
-    handleErrors(error);
+    handleErrors(error as string | string[]);
     return [];
   }
 }
+
+export async function addProduct(product: Product) {
+  try {
+    const data = await post(apiEndpoints.product, product);
+    return data as Game[];
+  } catch (error) {
+    handleErrors(error as string | string[]);
```

Comment:
I repeat again and again about repetitive logic. You have `try catch` for every block of code. 
It's bad practice because:
1. It's wasting time via copying/typing the try-catch block again and again. You should have global single handler in this case.
2. It increases main.js file significant
3. If I need to change try-catch logic here to get more details, for example, I need to spend 8+ hours to change all blocks in the huge project
4. Repetitive logic provides issue because the more code provides the greater chance of an error

For example `try catch` possible to place inside `post method` only once for the whole project